xbmc/filesystem/File.cpp

```diff
@@ -235,18 +235,11 @@ bool CFile::Open(const CURL& file, const unsigned int flags)
 
     CURL url(URIUtils::SubstitutePath(file));
 
-    if (!(m_flags & READ_NO_CACHE))
+    if (!(m_flags & READ_NO_CACHE) && (m_flags & READ_CACHED))
```

Comment:
I'd change whis one to only check for READ_CACHED and add an assert((m_flags & READ_NO_CACHE) && (m_flags & READ_CACHED)) as that seems like a broken scenario that should be fixed.